lib/src/client/dio_interceptor.dart

```diff
@@ -0,0 +1,107 @@
+  })  : _signatureBodySourceBuilder =
+            signatureBodySourceBuilder ?? SignatureBodySourceBuilder(),
+        _signatureBodySigner = signatureBodySigner!;
+
+  @override
+  Future onRequest(
+      RequestOptions options, RequestInterceptorHandler handler) async {
+    // Sign the request before it is sent
+    await _signRequest(options);
+    return handler.next(options); // Proceed with the request
+  }
+
+  /// Signs the outgoing request with a request signature.
+  Future<void> _signRequest(RequestOptions options) async {
+    options.headers
+        .remove(_options.headerName); // Remove existing signature header
+
+    final signatureBodySourceComponents =
+        _options.signatureBodySourceComponents.isNotEmpty
+            ? _options.signatureBodySourceComponents
+            : DefaultConstants.signatureBodySourceComponents;
+
+    Uint8List? body;
+    // Encode request body if required by signature components
+    if (options.data != null &&
+        signatureBodySourceComponents
+            .contains(SignatureBodySourceComponents.body)) {
+      body = utf8.encode(options.data.toString());
+    }
+
+    // Build parameters for constructing the signature body
+    final signatureBodySourceParameters = SignatureBodySourceParameters(
+        options.method,
+        options.uri,
+        options.headers
+            .map((key, value) => MapEntry(key.toString(), value.toString())),
+        _genGuid(),
+        _epochTime(),
+        _options.clientId!,
+        signatureBodySourceComponents,
+        body: body);
+
+    // Build the signature body source based on parameters
+    final signatureBodySource =
+        await _signatureBodySourceBuilder.build(signatureBodySourceParameters);
+
+    // Construct parameters for creating the signature
+    final signatureBodyParameters =
+        SignatureBodyParameters(signatureBodySource, _options.clientSecret!);
+
+    // Generate the signature using the signer
+    final signature = await _signatureBodySigner.sign(signatureBodyParameters);
+
+    // Format the signature header based on the specified pattern
+    final signatureHeader = _options.signaturePattern
+        .replaceAll("{ClientId}", _options.clientId!)
+        .replaceAll("{Nonce}", signatureBodySourceParameters.nonce)
+        .replaceAll(
+            "{Timestamp}", signatureBodySourceParameters.timestamp.toString())
+        .replaceAll("{SignatureBody}", signature);
+
+    // Add the signature header to the request headers
+    options.headers[_options.headerName] = signatureHeader;
+  }
+
+  // Generate a unique nonce
+  String _genGuid() {
+    final uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
+    return uuid.replaceAllMapped(RegExp('[xy]'), (match) {
+      final rand = DateTime.now().millisecond;
+      final index = match.group(0) == 'x' ? rand : (rand & 0x3 | 0x8);
+      return index.toRadixString(16);
+    });
+  }
+
+  // Get the current timestamp in seconds since epoch
+  int _epochTime() {
```

Comment:
This method is not equivalent to its .NET homologue. It's missing the clockSkew feature.